src/app/qgsdecorationcopyright.cpp

```diff
@@ -95,6 +97,11 @@ void QgsDecorationCopyright::render( const QgsMapSettings &mapSettings, QgsRende
   //Large IF statement to enable/disable copyright label
   if ( enabled() )
   {
+    QgsExpressionContext expressionContext = QgsExpressionContext();
```

Comment:
Can you use the expression context from the render context given here? Then you'll have access to the scale/rotation/etc variables

Reply:
@nyalldawson , yep, good idea.